test/unit/discovery_rule_test.rb

```diff
@@ -1,20 +1,59 @@
 require 'test_helper'
 
 class DiscoveryRuleTest < ActiveSupport::TestCase
+  setup do
+    @hg_org = FactoryGirl.create(:organization, :name => 'hgorg')
+    @hg_org.save!
+    @hostgroup = FactoryGirl.create(:hostgroup)
+    @hostgroup.organizations << @hg_org
```

Comment:
maybe add this to the line above with `:organizations => [@hg_org]` and then you won't need to save again?